src/main/scala/scalaz/stream/Process.scala

```diff
@@ -451,8 +451,18 @@ sealed trait Process[+F[_], +O]
 
   }
 
-
-
+  final def uncons[F2[x] >: F[x], O2 >: O](implicit F: Monad[F2], C: Catchable[F2]): F2[(O2, Process[F2, O2])] = step match {
+    case Step(head, next) => head match {
+      case Emit(as) => as.headOption.map(x =>
+        F.point[(O2, Process[F2,O2])]((x, Process.emitAll[O2](as drop 1) +: next))) getOrElse
+        C.fail[(O2, Process[F2,O2])](new scala.NoSuchElementException)
```

Comment:
@pchlupacek Done! I opted for `F2[Option[(O2, Process[F2, O2])]]` instead of the exact signature you suggested because I don't think it is relevant to return the same process in the event that it has ended for wtv reason.

I also opted to fail the `F` if the `Process` has failed.

What do you think?